src/Sylius/Bundle/ApiBundle/DependencyInjection/Compiler/SyliusLegacyAliasesPass.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+final class SyliusLegacyAliasesPass implements CompilerPassInterface
+{
+    public const NEW_PREFIXES_TO_LEGACY_MAP = [
+        'Sylius\Bundle\ApiBundle\Serializer' => 'Sylius\PriceHistoryPlugin\Infrastructure\Serializer',
+        'Sylius\Bundle\ApiBundle\Validator' => 'Sylius\PriceHistoryPlugin\Infrastructure\Validator',
+    ];
+
+    public function process(ContainerBuilder $container): void
+    {
+        foreach ($container->getDefinitions() as $serviceName => $definition) {
+            if (!$this->shouldHaveLegacyAlias($container, $serviceName)) {
+                continue;
+            }
+
+            $this->addLegacyAlias($container, $serviceName);
+        }
+    }
+
+    private function shouldHaveLegacyAlias(ContainerBuilder $container, string $serviceName): bool
+    {
+        foreach (self::NEW_PREFIXES_TO_LEGACY_MAP as $newPrefix => $legacyPrefix) {
+            $legacyServiceName = $this->getLegacyServiceName($serviceName);
+
+            if (str_starts_with($serviceName, $newPrefix) && !$container->has($legacyServiceName)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
```

Comment:
```suggestion
    private function shouldHaveLegacyAlias(ContainerBuilder $container, string $serviceName): bool
    {
        $results = array_filter(array_keys(self::NEW_PREFIXES_TO_LEGACY_MAP, fn(string $namespace) => str_contains($namespace, $serviceName);
});
    }
```
Not sure, but I think this code should also do the same work

Reply:
I think it may give false positives, as it will return true if there are any keys in `NEW_PREFIXES_TO_LEGACY_MAP` that contain `$serviceName`, even if the corresponding legacy service name exists in the container.
We probably could fix that by just checking also if the container has this service, but I would like to stay with the first method as looks for me more accurate